optuna/distributions.py

```diff
@@ -295,6 +295,50 @@ def _contains(self, param_value_in_internal_repr):
         return self.low <= value <= self.high
 
 
+class IntLogUniformDistribution(IntUniformDistribution):
+    """A uniform distribution on integers in the log domain.
+
+    This object is instantiated by :func:`~optuna.trial.Trial.suggest_int`, and passed to
+    :mod:`~optuna.samplers` in general.
+
+    Attributes:
+        low:
+            Lower endpoint of the range of the distribution. ``low`` is included in the range.
+        high:
+            Upper endpoint of the range of the distribution. ``high`` is included in the range.
+        step:
+            A step for spacing between values.
+    """
+
+    def __init__(self, low, high, step=1):
+        # type: (int, int, int) -> None
+
+        super(IntLogUniformDistribution, self).__init__(low, high, step)
+
+        if low <= 0.0:
+            raise ValueError(
+                "The `low` value must be larger than 0 for a log distribution "
+                "(low={}, high={}).".format(low, high)
+            )
+
+        if (high - low) % step != 0:
+            raise ValueError(
+                "The `high` - `low` must be a multiple of `step`, but "
+                "low={}, high={}, step={}.".format(low, high, step)
+            )
+
+    def single(self):
+        # type: () -> bool
+
+        return self.low == self.high
+
+    def _contains(self, param_value_in_internal_repr):
+        # type: (float) -> bool
+
+        value = param_value_in_internal_repr
+        return (self.low <= value <= self.high) & (value % self.step == 0)
```

Comment:
`_contains` is also the same as https://github.com/optuna/optuna/pull/1201/files#r424421651